macos_defaults.py

```diff
@@ -79,14 +79,16 @@ class Actions:
 
     def selected_text() -> str:
         try:
-            return ui.focused_element().AXSelectedText
-        except Exception:
-            try:
-                # ui.focused_element() sometimes returns NoElement.
-                # https://github.com/talonvoice/talon/issues/480
+            selected_text = ui.focused_element().AXSelectedText
+            if not selected_text:
```

Comment:
Could consider "if selected_text := ..." and reversing the conditional, but just stylistic — whatever works for you.

Reply:
Thought about that, but I find that less readable in this case because `selected_text` has a meaning outside of the if statement. I find `if x := ...` to be best for defining a variable that will be used within the statement

Reply:
But you are using it?

Reply:
I'm using it outside of the if statement is the thing. If I were only using it _within_ the body of the if statement I would use `:=`.

Reply:
Oh, I missed the second part of your statement about inverting the conditional, sorry. Let me see what that would look like.

Reply:
Aha, not only is this cleaner, it reveals a logic bug:

```python
    def selected_text() -> str:
        try:
            # Note that we only use the selected text if it's non-empty, and fallback if it's empty, even though
            # it could be perfectly valid for the program to report an empty selection.
            #
            # This is because some partially-accessible applications incorrectly report empty selections sometimes,
            # and this can be quite bad depending on the use case. This could be customized if we knew the application
            # was fully trustworthy.
            #
            # See https://github.com/phillco/talon-axkit/issues/59/
            if selected_text := ui.focused_element().AXSelectedText:
                return selected_text
            else:
                return actions.next()
        except Exception:
            return actions.next()
```

As written with either style, if we fall back due to an empty selection, and the fallback itself raised as an exception, we would call `actions.next()` a second time. The try/catch should only be for harvesting the selected text.

Reply:
Unfortunately I don't to fix that with the new style ;) how's this?

```python
    def selected_text() -> str:
        try:
            selected_text = ui.focused_element().AXSelectedText
        except Exception:
            selected_text = None

        if selected_text:
            # Note that we only use the selected text if it's non-empty, and fallback if it's empty, even though
            # it could be perfectly valid for the program to report an empty selection.
            #
            # This is because some partially-accessible applications incorrectly report empty selections sometimes,
            # and this can be quite bad depending on the use case. This could be customized if we knew the application
            # was fully trustworthy.
            #
            # See https://github.com/phillco/talon-axkit/issues/59/
            return selected_text
        else:
            return actions.next()
```